lmod/lmod.py

```diff
@@ -361,4 +360,20 @@ class LmodTestJavaMemory(LmodTestBase):
     def assert_output(self):
         return sn.assert_found(r'^True$', self.stdout)
 
+@rfm.simple_test
+class LmodTestCachedLoads(LmodTestBase):
+    descr += "load a module from spider cache and check for zero exit code"
+    executable = f'LMOD_CACHED_LOADS=1 ml foss/{calc_tcgen(-42)}'
+
+    @sanity_function
+    def assert_zero_exitcode(self):
+        return sn.assert_eq(self.job.exitcode, 0)
+
+@rfm.simple_test
+class LmodTestNotCachedLoads(LmodTestBase):
```

Comment:
why copy & paste all this? Just inherit from `LmodTestCachedLoads` and add a variable to insert for the LMOD_CACHED_LOADS value?

```
class LmodTestCachedLoads(LmodTestBase):
    LMOD_CACHED_LOADS = 1
    executable = f'LMOD_CACHED_LOADS={self.LMOD_CACHED_LOADS} ml foss/{calc_tcgen(-42)}'


class LmodTestNotCachedLoads(LmodTestCachedLoads):
    LMOD_CACHED_LOADS = 0